docker-compose.test.yml

```diff
@@ -0,0 +1,70 @@
+version: "2"
+services:
+
+  redis:
+    container_name: redis
+    image: redis:6.0.3
+    ports:
+      - 6379:6379
+    volumes:
+      - redis_data_test:/data
+
+  memcached:
+    container_name: memcached
+    image: memcached:1.6.6
+    ports:
+      - 11211:11211
+
+  pg:
+    container_name: pg
+    image: postgis/postgis:12-3.0
+    environment:
+      POSTGRES_USER: 'inaturalist'
+      POSTGRES_PASSWORD: 'inaturalist'
+      POSTGRES_DB: inaturalist_test
+    ports:
+      - 5432:5432
+    volumes:
+      - pg_data_test:/var/lib/postgresql/data
+      - ./schema/database.sql:/docker-entrypoint-initdb.d/database.sql
+
+  es:
+    container_name: es
+    image: docker.elastic.co/elasticsearch/elasticsearch:8.9.1
+    environment:
+      - discovery.type=single-node
+      - xpack.security.enabled=false
+    ports:
+      - 9200:9200
+    volumes:
+      - es_data_test:/usr/share/elasticsearch/data
+    command: >
+      /bin/sh -c "bin/elasticsearch-plugin list | grep -q analysis-kuromoji
+      || bin/elasticsearch-plugin install analysis-kuromoji;
+      /usr/local/bin/docker-entrypoint.sh"
+
+  api-test:
+    container_name: api-test
+    build: .
+    environment:
+      NODE_ENV: test
+      INAT_DB_HOST: pg
+      INAT_DB_USER : 'inaturalist'
+      INAT_DB_PASS: 'inaturalist'
+      INAT_ES_HOST: es
+      INAT_REDIS_HOST: redis
+      INAT_WEB_HOST: host.docker.internal
+      INAT_DB_NAME: inaturalist_test
+      INAT_ES_INDEX_PREFIX: test
+      DB_ALREADY_INITIALIZED: true
```

Comment:
Why not make sure this container has a psql-client package on it rather than having Docker initialize the DB for tests differently than other setups? I tested it locally with removing this ENV variable and installing `postgresql-client-11` on the api-test container and that worked OK. This shifts responsibility of setting up the DB in the Docker environment only, whereas if running locally native or using Github CI the process is different as its the responsibility of the test hooks in the codebase to set up the DB structure.

Reply:
Added a "test" target in the Dockerfile, that will install the postgresql-client